PlaybackLab/TVOSSwiftSampleApp/TVOSSwiftSampleApp/Players/FairplayPlayerViewController.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  FairplayPlayerViewController.swift
+//  TVOSSwiftSampleApp
+//
+//  Copyright © 2018 Ooyala. All rights reserved.
+//
+
+import Foundation
+
+class FairplayPlayerViewController: OOOoyalaTVPlayerViewController, OOEmbedTokenGenerator {
+  
+  var option: PlayerSelectionOption!
+  var pcode: String!
+  var playerDomain: String!
+  
+  // required for FairPlay.
+  var apiKey: String!
+  var secret: String!
+  // additionaly required if using OPT.
+  var authorizeHost: String!
+  var accountId: String!
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    self.pcode = self.option.pcode
+    self.playerDomain = self.option.domain
+    
+    // You need to fill the following params
+    self.authorizeHost = "http://player.ooyala.com"
+    self.apiKey = "Fill me in"
+    self.secret = "Fill me in"
+    self.accountId = "Fill me in"
+    
+    OODebugMode.setDebugMode(DebugMode.LogAndAbort)
+    assert(self.apiKey.contains(self.pcode), "self.pcode must be the long prefix of self.apiKey.")
+    
+    var options = OOOptions()
+    
+    // For this example, we use the OOEmbededSecureURLGenerator to create the signed URL on the client
+    // This is not how this should be implemented in production - In production, you should implement your own OOSecureURLGenerator
+    //   which contacts a server of your own, which will help sign the url with the appropriate API Key and Secret
+    options?.secureURLGenerator = OOEmbeddedSecureURLGenerator(apiKey: self.apiKey, secret: self.secret)
+    
+    self.player = OOOoyalaPlayer(pcode: self.pcode, domain: OOPlayerDomain(string: self.playerDomain), embedTokenGenerator: self, options: options)
+    
+    NotificationCenter.default.addObserver(self, selector: #selector(notificationHandler(_:)), name: nil, object: self.player)
+    
+    self.player.setEmbedCode(self.option.embedCode)
+    self.player.play()
+  }
+  
+  @objc func notificationHandler(_ notification: Notification) {
+    let name = notification.name
+    if name == NSNotification.Name.OOOoyalaPlayerTimeChanged {
+      return
+    }
+    
+    NSLog("Notification Received: %@. state: %@. playhead: %f", name.rawValue, OOOoyalaPlayer.playerState(toString: self.player.state()), self.player.playheadTime())
+  }
+  
+  func token(forEmbedCodes embedCodes: [Any]!, callback: OOEmbedTokenCallback!) {
+    var params = NSMutableDictionary()
+    
+    params["account_id"] = self.accountId
+
+    let uri = String(format: "/sas/embed_token/%@/%@", self.pcode, (embedCodes.map{$0 as! String}).joined(separator: ","))
```

Comment:
Try to cast _embedCodes_ from `[Any]!` to `[String]`

Reply:
Done